### PR TITLE
DOCS: Added The Longest Journey

### PR DESCRIPTION
**Pull request by Faalagorn**
~There are some *TODO* spots to fill yet as I have to find a retail copy of the game and I didn't try debugging commands.~ I have yet to complete the game on ResidualVM as well to confirm it's completable (with or without glitches, though some things are yet to be implemented, namely shadows and parts of UI). I wasn't sure if digital distribution versions (Steam and GOG) warrant a separate section and I have no idea if you can use iOS version of TLJ with ResidualVM. Keyboard shortcuts mentions in manual doesn't work yet and TLJ should be mentioned in http://www.residualvm.org/ as well, but I thought it's a good idea to let people know it's supported without looking down at code/commits.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -203,14 +204,46 @@ Space      | Skip cutscenes, interact
 Ctrl + c   | Force Quit (from command-line)
 Ctrl + q   | Quit (ingame)
 
+## 5. Running The Longest Journey
```
Comment (bgK) [inline README.md]: Please update the index at the top of the file with the new sections.

File: README.md [hunk 2]
```diff
@@ -203,14 +204,46 @@ Space      | Skip cutscenes, interact
...
+
+### 5.1. Required files ###
+
+You will need these elements from your game CDs or DVD :
+  * *TODO*
```
Comment (bgK) [inline README.md]: Please complete this TODO. This files on the CDs are the same as with the digital releases. For the 4-CD releases they can be copied as is. For the 2-CD/DVD versions some files need to be extracted from the installer.

File: README.md [hunk 3]
```diff
@@ -203,14 +204,46 @@ Space      | Skip cutscenes, interact
...
+  
+For digital distributions (Steam, GOG, demo from Steam or other sources),
```
Comment (bgK) [inline README.md]: For the Steam version at least, the font files are missing. They need to be obtained by some other way. For example from the demo version.

File: README.md [hunk 4]
```diff
@@ -338,12 +371,22 @@ infos       | Display the name of a node and show the associated scripts
...
+### 8.4. Debugging The Longest Journey ###
+
+The debug console can be triggered using Ctrl + d. Use the `help` command to
+display a list of the available commands.
+
+The most useful commands are:
+
+Command     | Description
+----------- | ------------------------------------------------------------
+*TODO*      | *TODO*
```
Comment (bgK) [inline README.md]: Remove this part if you don't want to document the commands. There is already some debugging documentation on the wiki anyways. No need to duplicate it here.

File: README.md [hunk 5]
```diff
@@ -203,14 +204,46 @@ Space      | Skip cutscenes, interact
...
+
+The mouse is used to interact with objects and menu elements.
+
+Available keyboard shortcuts:
+
+Key        | Binding
+---------- | ------------------------------
+Escape     | Skip video sequence or current line of dialogue, skip time if *Time Skip* option is enabled
+Alt + x    | Quit
+Alt + q    | Quit
```
Comment (bgK) [inline README.md]: Alt + x and Alt + q are platform specific. I'm not sure it's a good idea to document them here.

**Comment by Faalagorn** [comment]
Thanks a lot for the feedback! I'll address each of your notes separately and update the PR hopefully real soon based on the notes you provided :).

File: README.md (hunk 2 above)
Comment (Faalagorn) [reply README.md]: I was planning to update the CD/DVD if/when I find the game but Unfortunately I think I misplaced it somewhere (and I think I only had 4 CD version that might even had one CD scratched), but I'll add info based on what you said (and probably can do some googling as well).

File: README.md (hunk 3 above)
Comment (Faalagorn) [reply README.md]: I noticed the fonts are missing from both the Steam version and *Steam* demo (gotta check the demo from other sources) – the game works without them but the fancy fonts are indeed missing and ResidualVM mentions that, so guess it's worth adding. On the other hand it seems that Steam version is pre-patched with [patch 1.61](https://community.pcgamingwiki.com/files/file/345-the-longest-journey-patch/), but I'm not sure if there's any files that are related to ResidualVM as after extracting the EXE I can only see configurator tool and updated dll files inside, other than maybe some ini updates. Steam manual is updated for retail versions as well (mostly didn't mention CD or DVD requirements). On the other hand GOG include a Police Station fix, but I'm not sure if the version differences/patches are to be mention in this documentation?

File: README.md (hunk 4 above)
Comment (Faalagorn) [reply README.md]: I have copied it from the other games, assuming it's required – I can remove it, but I guess I can do some research as well to fill it up – are there any pointers to look at if I want to document it (assuming there are TLJ/Stark-exclusive debug commands there ofc :)).

File: README.md (hunk 5 above)
Comment (Faalagorn) [reply README.md]: Sorry, my bad – these are not platform specific shortcuts, but from the manual, just forgot to remove `Alt + x` and `Alt + q` after I realized they are not implemented in ResidualVM.

*However*, ESC is working quite fine, and with the settings now merged it properly doesn't work for time skipping (but still skips movies and dialogues) when the time skip isn't on, so I guess it's worth mentioning.

For reference, my original PR included (based on the manual):
```
Key        | Binding
---------- | ------------------------------
Escape     | Skip video sequence or current line of dialogue, skip time if *Time Skip* option is enabled
F1         | Diary Menu
F2         | Save game
F3         | Load game
F4         | Conversation Log
F5         | April's Diary (initially disabled)
F6         | Video replay
F7         | Game settings
F8         | Save a screenshot to your local game directory
F9         | Toggle subtitles on and off
F10        | Quit game and return to to main menu
A          | Cycle back through inventory cursor items
S          | Cycle forward through inventory cursor items
I          | Inventory
P          | Pause the game
X          | Display all exits on current location
Page Up    | Scroll up in dialogues and in your inventory
Up arrow   | Scroll up in dialogues and in your inventory
Page Down  | Scroll down in dialogues and in your inventory
Down arrow | Scroll down in dialogues and in your inventory
Enter      | Select currently highlighted dialogue choice
1 – 9      | Select a dialogue choice
Alt + x    | Quit
Alt + q    | Quit
```

File: README.md (hunk 3 above)
Comment (bgK) [reply README.md]: I usually use this demo version to get the fonts: https://archive.org/details/Longest_Journey_demo.

ResidualVM doesn't need any of the patches. I don't think it's needed to mention them.

File: README.md (hunk 4 above)
Comment (bgK) [reply README.md]: This wiki page has a list of useful debug commands: http://wiki.residualvm.org/index.php/TLJ_Debugging

The help command will also list all the commands that work with TLJ.

**Comment by Faalagorn** [comment]
I have updated everything now. There are some style changes I may want to try in a separate PR to improve particularly the debug section (some information is needlessly repeated and some options are common for multiple engines – namely `openlog`).

I had problem with describing some options, especially `dumpRoot` since it's not documented at the console/code or wiki (I didn't try tracking the commit that added it), so please take a look if the description are right – they are mostly pasted verbatim how wiki or usage hint says, as I wanted to avoid confusion by rephrasing it myself.

**Comment by bgK** [comment]
Thanks for your contribution! I merged your branch manually with some minor changes (dumpRoot console command description, added a paragraph mentioning installer archives).

Style changes are welcome if you want to do another PR.

**Pull request closed without merging**